### PR TITLE
Consolidate run and autorun buttons

### LINKED ISSUE #536: Improve UX of "Run" and "Autorun"
We should have a button

☐ Run

Clicking on "Run" gives you the usual running of programs. Clicking on ☐ leads to

☑ Auto

and autorun is turned on: Any edit that leads to a syntactically correct program will be run. The "Auto" would be not clickable.

Now, clicking on 

☑ 

toggles back to

☐ Run

### PR DESCRIPTION
**Pull request by jiayushe**
Fixes #536

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -78,6 +78,7 @@
   },
   "devDependencies": {
     "@blueprintjs/core": "^2.1.1",
+    "@blueprintjs/docs-theme": "^3.2.0",
```
Comment (jiachen247) [inline package.json]: sorry im a little confused! why are we adding a documentation dep in this pr?

File: src/components/workspace/ControlBar.tsx [hunk 2]
```diff
@@ -251,15 +256,28 @@ class ControlBar extends React.PureComponent<ControlBarProps, { value: string }>
...
+      </div>
+    ) : (
+      undefined
+    );
+
     return (
       <div className="ControlBar_editor pt-button-group">
-        {this.props.isRunning ? stopButton : this.props.isDebugging ? resetButton : runButton}
+        {toggleAutorunButton}
+        {this.props.isEditorAutorun
```
Comment (jiachen247) [inline src/components/workspace/ControlBar.tsx]: can we put this into function!! too must business logic for ternary operators.

File: src/components/workspace/ControlBar.tsx [hunk 3]
```diff
@@ -196,7 +197,11 @@ class ControlBar extends React.PureComponent<ControlBarProps, { value: string }>
           this.props.handleSetEditorSessionId!('');
         }
       };
-      xmlhttp.open('GET', 'https://' + LINKS.SHAREDB_SERVER + 'gists/' + this.state!.value, true);
+      xmlhttp.open(
+        'GET',
+        'https://' + LINKS.SHAREDB_SERVER + 'gists/' + this.state!.joinElemValue,
```
Comment (jiachen247) [inline src/components/workspace/ControlBar.tsx]: this should be externalized into another component leh we shouldn't be having logic in a dumb component such as the control bar.

Comment (martin-henz) [reply src/components/workspace/ControlBar.tsx]: The control bar turns out to be interesting. I think it should have the requested logic, in order to support run/autorun. I feel that run/stop/autorun can be succinctly supported in the requested feature. Let's merge this and see if we get feedback from users.

File: package.json (hunk 1 above)
Comment (jiayushe) [reply package.json]: Yea it is redundant, we have removed it.

File: src/components/workspace/ControlBar.tsx (hunk 3 above)
Comment (ccyccyccy) [reply src/components/workspace/ControlBar.tsx]: By externalized do you mean putting it into a helper file? If so we can do it in another PR.

Comment (jiachen247) [reply src/components/workspace/ControlBar.tsx]: yes I meant another file! if you think about it if the url changes lets say `gists` or `lastest` get changed to something else the control bar is the definitely wrong place to come to change things. As prof always says think in levels of abstraction!

Comment (jiachen247) [reply src/components/workspace/ControlBar.tsx]: I'll approve but at the rate we're going something is gonna break soon haha

**Pull request merged**